### ISSUE TITLE
Check before a version update

### ISSUE THREAD

**Issue by hxin** [statbio/Sargasso#68]
This issue is used to track the performance changes since version 1.1.2, in turns of accuracy and runtime. 

I used the hmr dataset and ran sargasso with --best/--conservative using the following three versions:
f2043d05d3683fe1962e57d9e363ba8717e1ed64|beforeXin
44798524024ff64b843e7299b36cfa541b01b327|master
462e27bea81f0b18967e7d41b619cb7ef6d5565a|masterWithIntron

The runtime was measured with sargasso running with one thread:
                                best         conservative
beforeXin               7:06:19     5:40:16
master                    7:07:10     5:50:39
masterWithIntron  7:07:58     6:00:58

![master-beforexin](https://user-images.githubusercontent.com/3407502/35816362-6a9ac734-0a92-11e8-8c96-0f68d164281f.png)
[master-beforeXin.txt](https://github.com/statbio/Sargasso/files/1695930/master-beforeXin.txt)
![master-masterwithintron](https://user-images.githubusercontent.com/3407502/35816363-6ad1b80c-0a92-11e8-960f-a487dc85a524.png)
[master-masterWithIntron.txt](https://github.com/statbio/Sargasso/files/1695931/master-masterWithIntron.txt)
![masterwithintron-beforexin](https://user-images.githubusercontent.com/3407502/35816364-6b0257e6-0a92-11e8-916e-2f1445a07263.png)
[masterWithIntron-beforeXin.txt](https://github.com/statbio/Sargasso/files/1695932/masterWithIntron-beforeXin.txt)

**Comment by hxin** [comment]
The results seem fine to me. The new code is slightly slower than before. Please let me know what you think @lweasel

**Comment by lweasel** [comment]
Yes, all looks good to me. Agree that it is worth having a look at some of the FPs.
